doc/rados/operations/bluestore-migration.rst

```diff
@@ -116,30 +116,62 @@ to evacuate an entire host in order to use it as a spare, then the
 conversion can be done on a host-by-host basis with each stored copy of
 the data migrating only once.
 
-#. Identify an empty host.  Ideally the host should have roughly the
-   same capacity as other hosts you will be converting (although it
-   doesn't strictly matter). ::
+First, you need have empty host that has no data.  There are two ways to do this: either by starting with a new, empty host that isn't yet part of the cluster, or by offloading data from an existing host that in the cluster.
 
-     NEWHOST=<empty-host-name>
+Use a new, empty host
+^^^^^^^^^^^^^^^^^^^^^
 
-#. Add the host to the CRUSH hierarchy, but do not attach it to the root::
+Ideally the host should have roughly the
+same capacity as other hosts you will be converting (although it
+doesn't strictly matter). ::
 
-     ceph osd crush add-bucket $NEWHOST host
+  NEWHOST=<empty-host-name>
 
-   If you would like to use an existing host that is already part of the cluster,
-   and there is sufficient free space on that host so that all of its data can
-   be migrated off, then you can instead do::
+Add the host to the CRUSH hierarchy, but do not attach it to the root::
 
-     ceph osd crush unlink $NEWHOST default
+  ceph osd crush add-bucket $NEWHOST host
 
-   where "default" is the immediate ancestor in the CRUSH map. (For smaller
-   clusters with unmodified configurations this will normally be "default", but
-   it might also be a rack name.) This will move the host out of the CRUSH
-   hierarchy and cause all data to be migrated off. Once it is completely empty of
-   data, you can proceed::
+Make sure the ceph packages are installed.
 
-     while ! ceph osd safe-to-destroy $(ceph osd ls-tree $NEWHOST); do sleep 60 ; done
+Use an existing host
+^^^^^^^^^^^^^^^^^^^^
 
+If you would like to use an existing host
+that is already part of the cluster, and there is sufficient free
```

Comment:
Does this mean the host is any non-OSD host(mon, client etc), Is it still applicable if the host has some osd id's and one wants to add additional osd's?

Reply:
I don't see why not

Reply:
rereading, i dont understand why 'whole host replacement' should use 'existing host' instead of 'new host', since this doc is talking about bluestore migration, there can be only 2 cases a) convert existing osd  b) add new hosts with osds.   so even 'existing host' should fall under b)

Reply:
If you have a 10 node cluster and no extra hardware, the best strategy is probably to (1) evacuate 1 host, then (2) do this host-by-host conversion for 9 hosts, then add the final empty host back in.